### PR TITLE
Add zfs cloning

### PR DESCRIPTION
**Pull request by wishdev**
Adds the ability to clone a snapshot when creating a zfs volume

### CODE REVIEW

**Comment by gregdek** [comment]
@johanwiren Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please give it a +1; at that point, we will merge.

**Comment by johanwiren** [comment]
I think that the "clone" option should be named "origin" to better follow zfs naming.

Also, please add an example to the doc in how to create and destroy clones.

File: system/zfs.py [hunk 1]
```diff
@@ -66,6 +66,10 @@
       - The checksum property.
     required: False
     choices: ['on','off',fletcher2,fletcher4,sha256]
+  clone:
+    description:
+      - Name of the snapshot to clone
+   required: False
```
Comment (johanwiren) [inline system/zfs.py]: Wrong indentation.

**Comment by wishdev** [comment]
@johanwiren - Switched the property name, cleaned up the identing issue and added examples. I only added a generic example for destroying a file system because there is nothing special about removing a cloned file system.

**Comment by johanwiren** [comment]
Great work!

@gregdek I approve.

**Pull request merged**